### ISSUE TITLE
MySQL 5.7+ support

### ISSUE THREAD

**Issue by ToBeReplaced** [piwik/piwik#6642]
Piwik makes use of the `IGNORE` clause in `ALTER TABLE`, which has been removed as of MySQL 5.7.

There may be other incompatibilities between piwik and MySQL 5.7. The deprecated and removed features can be found at https://dev.mysql.com/doc/refman/5.7/en/mysql-nutshell.html .

See #6623 for how the incompatibilities manifest in a new installation.

**Comment by mattab** [comment]
In saying to the community "Piwik supports Mysql 5.7" we should run our automated test suites on Mysql 5.7 in CI

**Comment by johsin18** [comment]
Would be great to fix this ASAP.  Usually you don't have control over the MySQL version your provider uses. And the Piwik system check does not even complain, so it's hard to find this problem.

**Comment by mattab** [comment]
@johsin18 did you notice any issue on Mysql 5.7 so far?

**Comment by johsin18** [comment]
Well, yeah, the actual problem is stated in issue 6623.  Effectively, no visits are recorded at all after a fresh install.

**Comment by mattab** [comment]
@johsin18 Good point, if we don't make Piwik compatible with Mysql 5.7, Piwik may fail without users knowing why which would be confusing and possibly frustrating. Adding to `Short term` and hopefully we can make progress on this

**Comment by Ramoonus** [comment]
bump

**Comment by mattab** [comment]
* Important; check issue https://github.com/piwik/piwik/issues/9131 for more information - this was marked as duplicate but has interesting info and bug  description

**Comment by otheus** [comment]
@mattab Look, I don't want to step on any toes here, but marking 9131 as a "duplicate" of this one seems to only contribute to confusion in the short- and long-run, especially since that bug isn't specifically referenced by this issue, yet is now marked as "closed". In the dev environments I've been in, these would be marked as related, or assigned the same tag ("mysql 5.7.5"). To the extreme counterexample, you could have an "issue" named "fix all bugs" and mark everything as a duplicate of it. ;)

**Comment by mattab** [comment]
> , especially since that bug isn't specifically referenced by this issue, yet is now marked as "closed"

the bug is referenced in my comment above + it appears in this page linked above. there is a small risk we would forget to look at #9131 but I trust whoever works on this will not miss it :+1: although, for sure it would work well to leave other issue opened too

**Comment by otheus** [comment]
I believe you missed my main point. These are separate issues. Marking it as "duplicate" is confusing and erroneous. For instance, existing installations that upgrade to MySQL 5.7.5 will work with the patch I present in #9131. If you want to create an "umbrella" issue, that's perfectly reasonable; but if you close the underlying issues as duplicates, it (1) makes the collective issue harder to attack; developers will try to pick off seemingly easier smaller problems than one, monolithic harder one; (2) makes this particular problem open-ended, since more issues may come to light after the known ones are uncovered, (3) is harder to track which individual issues still need resolving. 

Also: FWIW, this is the wrong "place" for this discussion: I just don't know of an alternative. The forums perhaps?

**Comment by tsteur** [comment]
I'm getting a bit confused by the duplicates as well and not sure what is to be done here. Finding it easier too to have separate issues for specific things. There's no advantage to me by merging issues into one big issue.

Is it like a general issue to install 5.7.5+ and to run all the tests?

**Comment by mattab** [comment]
Ok, re-opened the other one for now. 

> Is it like a general issue to install 5.7.5+ and to run all the tests?

sounds good :+1: 

Later on we could run all tests on 5.7.5 and maybe could create an issue for it

**Comment by tsteur** [comment]
I issued two pull requests for other existing issues: https://github.com/piwik/piwik/pull/9383 and https://github.com/piwik/piwik/pull/9382

Apart from that I couldn't find any problems with MySQL 5.7 (mysql  Ver 14.14 Distrib 5.7.9). I ran integration and system tests on my local instance. There were some system test failures but I think they are not related to MySQL 5.7. I also tried an installation + some simple tracking and it worked. I guess it's better to just see if any other error occurs and then have a look at the specific error when something comes up.

I'd suggest to close this issue
